### PR TITLE
Added LocalDomain support MeshService{}

### PR DESCRIPTION
**Pull request by brunoterkaly**
**Description**:
This PR is about adding ClusterDomain support to MeshService{}. Currently, it adds the concept of "local" cluster, unlocking multicluster functionality with Kubernetes. Multi-cluster is a strategy for deploying an application on or across multiple Kubernetes clusters with the goal of improving availability, isolation, and scalability.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [**X**] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

| Q: | A:  |
|:-------------------- |:--------------------  |
| Does this change contain code from or inspired by another project? | No  |
| Did you notify the maintainers and provide attribution? | No  |
| Is this a breaking change? | No  |

### CODE REVIEW

**Comment by brunoterkaly** [comment]
```
I.   Previous Comments in for this work
     A. Formatting - Ran the linter but ended up getting a massive amount
        of needed fixes outside the scope of my own work.
        1.  There may be some un-needed blank lines.
     B. Not all tests are passing as I believe they need to run
        separately from VS code. Those tests currently not passing include:
        1.  pkg/envoy/cds/response_test.go
        2.  tests/e2e/suite_test.go
        3.  tests/scale/suite_test.go
        4.  tests/scenarios/suite_test.go
```

**Review by snehachhabria — COMMENTED** [review]
For the DCO check to pass, you will need to sign your PR following the steps here : https://github.com/openservicemesh/osm/blob/main/CONTRIBUTING.md#sign-your-work

File: pkg/envoy/xdsutil.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ func ParseEnvoyServiceNodeID(serviceNodeID string) (*PodMetadata, error) {
 // GetLocalClusterNameForService returns the name of the local cluster for the given service.
 // The local cluster refers to the cluster corresponding to the service the proxy is fronting, accessible over localhost by the proxy.
 func GetLocalClusterNameForService(proxyService service.MeshService) string {
-	return GetLocalClusterNameForServiceCluster(proxyService.String())
+	return GetLocalClusterNameForServiceCluster(proxyService.NameWithoutCluster())
```
Comment (steeling) [inline pkg/envoy/xdsutil.go:295]: can you switch this back to string?

Comment (brunoterkaly) [reply pkg/envoy/xdsutil.go]: Fixed.

File: go.mod [hunk 2]
```diff
@@ -38,27 +39,22 @@ require (
 	github.com/prometheus/common v0.10.0
 	github.com/rs/zerolog v1.18.0
 	github.com/servicemeshinterface/smi-sdk-go v0.5.0
-	github.com/spf13/cobra v1.1.1
+	github.com/spf13/cobra v1.1.3
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/objx v0.3.0 // indirect
 	github.com/stretchr/testify v1.7.0
-	golang.org/x/sys v0.0.0-20210414055047-fe65e336abe0 // indirect
-	golang.org/x/tools v0.1.1-0.20210319172145-bda8f5cee399 // indirect
```
Comment (snehachhabria) [inline go.mod:46]: We need to have these indirect references, lets not remove them

**Review by snehachhabria — COMMENTED** [review]
Can we close this PR give that : https://github.com/openservicemesh/osm/pull/3642 is approved ?

**Comment by brunoterkaly** [comment]
Superceded by Sean's fork.

**Pull request closed without merging**